### ISSUE TITLE
Error in Inference? - Same output, Different inputs

### ISSUE THREAD

**Issue by rbommasanimozilla** [OpenNMT/OpenNMT-py#838]
Use Case: Summarization on dataset similar to CNN/DM (Newsroom dataset)
I have been training a variety of models for benchmarking hyperparameters. Training appears to proceed well, with my best models reach 27-30% accuracy and 80-100 perplexity on the validation set (Validation loss approximately monotonically declines).
Model: 2 layer BRNN encoder, 2 layer RNN decoder, copy_attention, most other parameters are default parameters in summarization tutorial. 

Problem: Across all inputs, the output at inference time is essentially, if not exactly, the same.

Command:
python translate.py -beam_size 5 -model models/newsroom-2048512x216250lr2_step_225000.pt -src data/cornell/newsroom/val.txt.srcp -output testout/newsroom-2048512x216250lr2_step_225000.out -min_length 35 -verbose -stepwise_penalty -coverage_penalty summary -beta 5 -length_penalty wu -alpha 0.9 -verbose -block_ngram_repeat 3 -ignore_when_blocking "." "</t>" "<t>"
*Tried beam sizes of 1,5,10,25; beta values of 1,5; alpha values of 0.5, 0.7, 0.9
**File used intentionally ends in .srcp, has been preprocessed identically to preprocessing used by See et al for their paper on pointer-generator networks with coverage for summarization.

Output:
SENT 80: ('st.', 'george', ',', 'utah', '-', 'at', '7:20', 'a.m.', ',', 'when', 'the', 'hikers', 'gather',  ..)
PRED 80: collection of all usatoday.com coverage of <unk> <unk> , including articles , videos , photos , and quotes . commentary and archival information about <unk> , including articles , videos , photos , and quotes .
PRED SCORE: -6.0387

SENT 81: ('salesforce', 'ceo', 'marc', 'benioff', 'took', 'home', 'less', 'money', 'in', ...)
PRED 81: collection of all usatoday.com coverage of <unk> <unk> , including articles , videos , photos , and quotes . commentary and archival information about <unk> , including articles , videos , photos , and quotes .
PRED SCORE: -7.3252

SENT 82: ('toto', ',', 'i', "'ve", 'a', 'feeling', 'we', "'re", 'not', 'in', 'alicia-ville', 'anymore.i', "'m", 'betting', ...)
PRED 82: collection of all usatoday.com coverage of <unk> <unk> , including articles , videos , photos , and quotes . commentary and archival information about <unk> , including articles , videos , photos , and quotes .
PRED SCORE: -7.9007

Things I have tried:
1. Varying almost all beamsearch params/batch_size/model hyperparams/etc.
2. Decoding on Single GPU (with and without -fast flag) and on CPU - Suggests the problem should not be in collapsing beam and batch dimensions/subsequent resolution.
3. Training a model to copy (output seq = input seq) - Model converges to 100% accuracy and performs well at inference, suggesting neither training nor inference is completely broken.
Any help would be greatly appreciated!

**Comment by rbommasanimozilla** [comment]
@vince62s have you recently done inference for models/run into any of these problems. Still really haven't narrowed down why this problem is arising, but I am fairly certain it is on the inference side and I don't see any preprocessing bugs/problems with the data.

**Comment by vince62s** [comment]
what kind of model did you train ?
30% accuracy is very poor as well is 100 PPL.

**Comment by rbommasanimozilla** [comment]
Dataset: Newsroom: 4GB Train, 400mb val, 400mb test - 1 million training examples
Model: Basically 2 layer brnn encoder, 2 layer decoder, uses copy attention
All of my parameters are the same/very similar to those given in the summarization tutorial.

**Comment by vince62s** [comment]
not sure, maybe @pltrdy may advise.

**Comment by rbommasanimozilla** [comment]
On current master it looks like this is fixed, I am currently getting over 60% accuracy on newsroom (training to about 9.00 ppl) and 50% accuracy (training to about 14 ppl) on cnn/dm. Not sure what caused this. 
Thanks for helps @vince62s
